[ldot-bdot] Misc clarifications on RVBN Algorithm (RVBNA) (#11)
aswaterman/riscv-misc: isa/ldot-bdot/ldot-bdot.adoc

`o` is defined to accommodate any carry overflow and is defined to be `ceil(log2(n))`.

The number of guard bits, `g`, is defined to be `ceil(log2(n))`.

Note:: It is expected that `n` would be a constant defined by each operation/implementation using RVBNA.

> It will not depend on a dynamic parameter such as `vl`. Dot product operations are still allowed to use `vl` or any dynamic length parameter but are encouraged to not link this to the bulk normalization parameters (which is similar to assuming identity-value products are injected in place of inactive elements).

Author: ```suggestion
It is not expected to depend on a dynamic parameter such as `vl`. Dot product operations are still allowed to use `vl` or any dynamic length parameter but are encouraged to not link this to the bulk normalization parameters (which is similar to assuming identity-value products are injected in place of inactive elements).
```

`n` may vary depending on the element width (so have a dependence on `SEW` for example).

> Note:: Given the recommendation made on `n`, RVBNA mandates a fixed value of `g` for a given format size and multiple value of `g` if `n` varies with the format size. This reduces the space of possible implementations.

Author: ```suggestion
Note:: Given the recommendation made on `n`, RVBNA mandates a fixed value of `g` for a given format size. This reduces the space of possible implementations. An implementation using RVBNA for different formats with different `n` values requires support for multiple values of `g`.
```

> Supporting multiple values of `g` is straightforward: an implementation can build a shared accumulator datapath based on the maximum `gmax` value.

Author: ```suggestion
Supporting multiple values of `g` is straightforward: an implementation can build a shared accumulator datapath based on the maximum `g` value: `gmax`.
```

> Behavior for smaller `g` values would be obtained by OR-reducing both the sticky bits defined by `gmax` and the extra `gmax - g` least significant bits of the aligned products into the least significant accumulator bit. This should not require more than a few extra gates.

Author: ```suggestion
Behavior for smaller `g` values would be obtained by OR-reducing both the sticky bits defined by `gmax` and the extra `gmax - g` least significant bits of the aligned products into the least significant accumulator bit. This should not require more than a few extra gates for hardware implementations.
```

Note:: Any one of the `2*p` bits of the max product can be the leading bit due to leading zeros in subnormal inputs. It is also possible for the maximum product to have more leading zeros than the other products (including when product alignment is taken into account). This is discussed in more detailed in the next section <<#RVBNAReferenceExponent>>.

> Note:: The RVBNA pseudocode assumes that operands are IEEE encoded, but this is not actually a requirement. The pseudocode can easily be generalized to other floating-point encodings (e.g. Open Compute OFP formats).

Author: ```suggestion
Note:: The RVBNA pseudocode assumes that operands are IEEE encoded, but this is not actually a requirement of RVBNA. The pseudocode can easily be generalized to other floating-point encodings (e.g. Open Compute OFP formats).
```
